src/nvim/ex_cmds.c

```diff
@@ -4960,3 +4961,27 @@ void ex_oldfiles(exarg_T *eap)
     }
   }
 }
+
+void ex_trust(exarg_T *eap)
+{
+  char *p = skiptowhite(eap->arg);
+  size_t len = (size_t)(p - eap->arg);
+  char *arg1 = xcalloc(len + 1, sizeof(char));
+  memcpy(arg1, eap->arg, len);
+
+  char *errmsg;
+  if (STRNCMP(arg1, "++deny", len) == 0) {
```

Comment:
This is incomplete. I'm assuming `bufnr` for now, because I wanted to confirm how we want to handle distinguishing between `bufnr` and `path` arguments. I was thinking about the following:
- if something is a string, then it's `path`
- if something is a number, then it's `bufnr`
- user can wrap the argument in quotes (e. g. `'0'`) to have it interpreted as a path
Would this be ok?

Reply:
Actually this can be parsed for you with proper flags set in `ex_cmds.lua`. `COUNT` flag accepts count/number, `ARGOPT` flag accepts `++opts`, `addr_type='ADDR_BUFFERS'` to set the count/number type

Reply:
Yep, I based this off of `:autocmd`.

I just want to confirm the approach here:
I'd have to:
- parse the argument inside `getargopt()` in `ex_docmd.c`
- add the corresponding flag to `exarg_T`
- add information about the flag to `:h ++opt`?
It seems slightly too generic for a very specific case - even having a special flag in `exarg_T` to tell me if I should `deny` or `remove` feels off to me and makes me wonder if that's the reason `:autocmd` handles this internally?

Reply:
> This is incomplete. I'm assuming `bufnr` for now, because I wanted to confirm how we want to handle distinguishing between `bufnr` and `path` arguments. I was thinking about the following:
> 
> * if something is a string, then it's `path`
> * if something is a number, then it's `bufnr`
> * user can wrap the argument in quotes (e. g. `'0'`) to have it interpreted as a path
>   Would this be ok?

This seems reasonable and matches other commands which take either a buffer number or name (e.g. `:bdelete`). Maybe take a peek at how that command handles this and see if we can re-use anything there.

Reply:
Sorry, I think not using ARGOPT for this is fine, doing this like `:autocmd` should be okay.

Actually now that I think about it, maybe we shouldn't allow buffer numbers inside the command at all. Either file path or no arguments for the current buffer.

Reply:
>Actually now that I think about it, maybe we shouldn't allow buffer numbers inside the command at all. Either file path or no arguments for the current buffer.

I also agree with this, actually.

Reply:
Okay, so I will update this to be `:trust [++deny] [++remove] [{file}]` and `trust(action, filepath|nil)`.

Reply:
This is just about `:trust`, lua function can do more things, and should accept a buffer number

Reply:
You're right. My bad :) I'll leave lua function as is.

Reply:
Updated `:trust` in f3dac4b4a65bfd99ce965ee48217f0ce73389ded